src/Core/Managed/Shared/Channel/Transmission.cs

```diff
@@ -212,6 +233,112 @@ public virtual Task<HttpWebResponseWrapper> SendAsync()
         }
 #endif // !NET40
 
+        /// <summary>
+        /// Splits the Transmission object into two pieces using a method 
+        /// to determine the length of the first piece based off of the length of the transmission.
+        /// </summary>
+        /// <returns>
+        /// A tuple with the first item being a Transmission object with n ITelemetry objects
+        /// and the second item being a Transmission object with the remaining ITelemetry objects.
+        /// </returns>
+        public virtual Tuple<Transmission, Transmission> Split(Func<int,int> calculateLength)
+        {
+            Transmission transmissionA = this;
+            Transmission transmissionB = null;
+
+            // We can be more efficient if we have a copy of the telemetry items still
+            if (this.TelemetryItems != null)
+            {
+                // We don't need to deserialize, we have a copy of each telemetry item
+                int numItems = calculateLength(this.TelemetryItems.Count);
+                if (numItems != this.TelemetryItems.Count)
+                {
+                    List<ITelemetry> itemsA = new List<ITelemetry>();
+                    List<ITelemetry> itemsB = new List<ITelemetry>();
+                    var i = 0;
+                    foreach (var item in this.TelemetryItems)
+                    {
+                        if (i < numItems)
+                        {
+                            itemsA.Add(item);
+                        }
+                        else
+                        {
+                            itemsB.Add(item);
+                        }
+                        i++;
+                    }
+
+                    transmissionA = new Transmission(
+                        this.EndpointAddress,
+                        itemsA);
+                    transmissionB = new Transmission(
+                        this.EndpointAddress,
+                        itemsB);
+                }
+            }
+            else if (this.ContentType == JsonSerializer.ContentType)
+            {
+                // We have to decode the payload in order to split
+                bool compress = this.ContentEncoding == JsonSerializer.CompressionType;
+                string[] payloadItems = JsonSerializer
+                    .Deserialize(this.Content, compress)
+                    .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
This deserialization code is in a couple places, is there an easy way to only have one copy?